javaslang-benchmark/src/test/java/javaslang/benchmark/PriorityQueueBenchmark.java

```diff
@@ -0,0 +1,135 @@
+package javaslang.benchmark;
+
+import org.junit.Test;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.*;
+import scala.math.Ordering;
+import scala.math.Ordering$;
+import scalaz.*;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static javaslang.benchmark.BenchmarkResultAggregator.displayRatios;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 3, time = 1, timeUnit = TimeUnit.SECONDS)
+@Fork(value = 2, jvmArgsAppend = { "-XX:+UseG1GC", "-Xss100m", "-Xms1g", "-Xmx1g", "-ea" })
+public class PriorityQueueBenchmark {
+    @Param({ "10", "1000", "100000" })
+    protected int CONTAINER_SIZE;
+
+    @Test
+    public void launchBenchmark() throws Exception {
+        final Options opts = new OptionsBuilder()
+                .include(PriorityQueueBenchmark.class.getSimpleName())
+                .shouldDoGC(false)
+                .shouldFailOnError(true)
+                .build();
+
+        displayRatios(new Runner(opts).run());
+    }
+
+    private List<Integer> ELEMENTS;
+
+    @Setup
+    public void setup() {
+        final Random random = new Random(0);
+
+        final List<Integer> copy = new ArrayList<>(CONTAINER_SIZE);
+        for (int i = 0; i < CONTAINER_SIZE; i++) {
+            final int value = random.nextInt(CONTAINER_SIZE) - (CONTAINER_SIZE / 2);
+            copy.add(value);
+        }
+
+        ELEMENTS = Collections.unmodifiableList(copy);
+        assert ELEMENTS.size() == CONTAINER_SIZE;
+    }
+
+    @Benchmark
+    @SuppressWarnings("Convert2streamapi")
+    public void sort_java(Blackhole bh) {
```

Comment:
would be great to have also `mutable` and `immutable` as part of the test names.